examples/src/java/org/pantsbuild/example/README.md

```diff
@@ -205,6 +205,37 @@ This generates a zipfile with runnable contents; instead of a zipfile, we could
 contents a directory tree, a giant jar, or something else.
 <a pantsref="jvm_bundles">Learn more about bundles</a>.
 
+Compiler Options
+----------------
+
+Compiler options are set the same way any options are set. Please remember to prefix the flag so
```

Comment:
Adding documentation on this is great! However I'm not sure this opening sentence is particularly enlightening, and in fact may be misleading. The "compiler option sets" mechanism adds extra indirection to the idiomatic use of Pants options.

It might be better to explain what "compiler option sets" are, and how the config setting and the per-target settings interact. And what "enabled args" and "disabled args" mean, and then give examples. And only then remind people that `-S` and `-C` are zinc prefixes to indicate passthrus to the underlying compilers.

Also, would be good to explain how to set the repo-wide defaults for the target settings (is that even possible? I don't know myself)

My point is that this is a powerful and poorly-documented feature, so your instinct to document it is absolutely on point. But let's be a lot more verbose and detailed here...